Warn that overriding `__builtins__` for `eval` is not a security mechanism (#145773)
python/cpython: Doc/library/functions.rst

   .. warning::

      This function executes arbitrary code. Calling it with

>       user-supplied input will lead to security vulnerabilities.

Reviewer: ```suggestion
      un-trusted user-supplied input can easily lead to security vulnerabilities.
```

Author: ```suggestion
      untrusted user-supplied input will lead to security vulnerabilities.
```

   The *source* argument is parsed and evaluated as a Python expression
   (technically speaking, a condition list) using the *globals* and *locals*
   mappings as global and local namespace.  If the *globals* dictionary is
   present and does not contain a value for the key ``__builtins__``, a
   reference to the dictionary of the built-in module :mod:`builtins` is
   inserted under that key before *source* is parsed.
   Overriding ``__builtins__`` can be used to restrict or change the available

>    names, but this is **not** a security mechanism: the executed code can

Reviewer 2: Either decide to keep the "this" or drop it, but make it similar to exec().

Author: I have no idea what happened. The suggestions I applied had "this" in both, GH must have some outtage.

>    names, but this is **not** a security mechanism: the executed code can

Reviewer 2: ```suggestion
   names, but this is **not** a security mechanism: the evaluated code can
```

For `eval`, let's call it "evaluated" code rather than "executed" code.

Reviewer 2: NVM, we use "executed" afterwards. It's only used in the preamble of the function.

>    still access builtins.

Reviewer: ```suggestion
   names, but this is **not** a security mechanism: the executed code can
   still access all builtins.
```

   .. warning::

      This function executes arbitrary code. Calling it with

>       user-supplied input will lead to security vulnerabilities.

Reviewer: ```suggestion
      un-trusted user-supplied input can easily lead to security vulnerabilities.
```

Author: We are generally trying to strengthen the warning (we held a little vote, people are in favour), but IMO this suggestion reverts that. Might I ask why exactly you want it to be so?

Reviewer: I understand the desire to strengthen the warning, I am in favor of that. I think it's important to introduce the idea of "un-trusted" input, and I don't want to get into arguments with people that point out (as @picnixz did) where statements are incorrect.

I don't know where the vote was held. I will defer to security experts.  Can we keep "un-trusted"?

Author: ```suggestion
      untrusted user-supplied input will lead to security vulnerabilities.
```

   If the *globals* dictionary does not contain a value for the key
   ``__builtins__``, a reference to the dictionary of the built-in module
   :mod:`builtins` is inserted under that key.
   Overriding ``__builtins__`` can be used to restrict or change the available
   names, but it is **not** a security mechanism, the executed code can

>    still access builtins.

Reviewer: ```suggestion
   names, but is **not** a security mechanism: the executed code can
   still access all builtins.
```
